### PR TITLE
Comments export import

### PR DESCRIPTION
**Pull request by aapomm**
### Spec
We want to have the ability to export/import comments in Projects to be able to migrate the discussion along with the findings.

#### Proposed Solution
Add builders and parsers in dradis-projects for Comments

### How to test
*Note: this PR is more easily tested once there's a way to view comments*
1. Create a project
2. Add an issue
3. Add comments to the issue
4. Export the project
5. Create a new project(2)
6. Upload the export from Step 4 to project(2).
7. Assert that the comments were migrated.

### CODE REVIEW

File: lib/dradis/plugins/projects/export/v1/template.rb [hunk 1]
```diff
@@ -16,6 +16,18 @@ def build_activities_for(builder, trackable)
       end
     end
 
+    def build_comments_for(builder, commentable)
+      builder.comments do |comments_builder|
+        commentable.comments.each do |comment|
+          comments_builder.comment do |comment_builder|
+            comment_builder.content(comment.content)
```
Comment (etdsoft) [inline lib/dradis/plugins/projects/export/v1/template.rb]: this probably needs to go through CDATA

File: lib/dradis/plugins/projects/upload/v1/template.rb [hunk 2]
```diff
@@ -55,6 +55,20 @@ def create_activity(trackable, xml_activity)
         validate_and_save(activity)
       end
 
+      def create_comments(commentable, xml_comments)
```
Comment (etdsoft) [inline lib/dradis/plugins/projects/upload/v1/template.rb]: this is definitely a different version of the template, not `v1`

File: lib/dradis/plugins/projects/upload/v1/template.rb [hunk 3]
```diff
@@ -105,7 +119,7 @@ def finalize_evidence
       # Fix relationships between nodes to ensure parents and childrens match
       # with the new assigned :ids
       def finalize_nodes
-        pending_changes[:orphan_nodes].each do |node|
+        pending_changes[:orphan_nodes].each_with_index do |node, i|
```
Comment (etdsoft) [inline lib/dradis/plugins/projects/upload/v1/template.rb]: why is this required?

File: lib/dradis/plugins/projects/upload/v1/template.rb [hunk 4]
```diff
@@ -55,6 +55,19 @@ def create_activity(trackable, xml_activity)
         validate_and_save(activity)
       end
 
+      def create_issue(issue, xml_issue)
```
Comment (dormi) [inline lib/dradis/plugins/projects/upload/v1/template.rb]: what is the purpose of this change in the old template?

Comment (aapomm) [reply lib/dradis/plugins/projects/upload/v1/template.rb]: I wanted to have a method that I can override in v2. Of course, I can just override the original method `parse_issues` but it seemed excessive.

File: lib/dradis/plugins/projects/upload/v2/template.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+              "\n\nThis comment was imported into the project. Original author: "\
+              "#{author_email}."
+          end
+
+          return false unless validate_and_save(comment)
+        end
+      end
+
+      def create_issue(issue, xml_issue)
+        # TODO: Need to find some way of checking for dups
```
Comment (etdsoft) [inline lib/dradis/plugins/projects/upload/v2/template.rb]:
```
return false unless super
return false unless create_comments(issue, xml_issue.xpath('comments/comment'))
```

File: lib/dradis/plugins/projects/upload/v2/template.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            commentable_id: commentable.id,
+            commentable_type: commentable.class.to_s,
+            content: xml_comment.at_xpath('content').text,
+            created_at: Time.at(xml_comment.at_xpath('created_at').text.to_i),
+            user_id: user_id_for_email(author_email)
+          )
+
+          if comment.user.email != author_email
+            comment.content = comment.content +
+              "\n\nThis comment was imported into the project. Original author: "\
```
Comment (etdsoft) [inline lib/dradis/plugins/projects/upload/v2/template.rb]: `Original author not available in this Dradis instance:`
